packages/terra-application-layout/src/menu/RoutingMenu.jsx

```diff
@@ -0,0 +1,159 @@
+   * @param {String} path is the path to be matched against
+   * @param {Array} menuItems is the Array of menuItem objects as specified by the RoutingMenu's proptype definition.
+   */
+  static getSelectedChildKey(path, menuItems) {
+    for (let i = 0, itemCount = menuItems.length; i < itemCount; i += 1) {
+      const navItem = menuItems[i];
+      if (navItem.path && matchPath(path, { path: navItem.path })) {
+        return navItem.path;
+      }
+    }
+    return undefined;
+  }
+
+  constructor(props) {
+    super(props);
+
+    this.handleMenuChange = this.handleMenuChange.bind(this);
+
+    /**
+     * The menuItems are checked against the current location at initialization to ensure the
+     * the desired selection styles are present.
+     */
+    this.state = {
+      selectedChildKey: RoutingMenu.getSelectedChildKey(props.location.pathname, props.menuItems),
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
+    /**
+     * The selectedChildKey is re-evaluated when new props are received to keep the internal state consistent.
+     */
+    this.setState({
+      selectedChildKey: RoutingMenu.getSelectedChildKey(nextProps.location.pathname, nextProps.menuItems),
+    });
+  }
+
+  handleMenuChange(event, data) {
+    const { routingStackDelegate, layoutConfig } = this.props;
+
+    this.setState({
+      selectedChildKey: data.selectedChildKey,
+    });
+
+    /**
+     * If the menuItem does not indicate the presence of a subsequent menu, it is assumed that a terminal
+     * menu item has been selected. If the `toggleMenu` function is defined on the layoutConfig, then it is called
+     * to close the menu before navigating.
+     */
+    if (!data.metaData.hasSubMenu && layoutConfig.toggleMenu) {
+      return layoutConfig.toggleMenu().then(() => {
+        routingStackDelegate.show({ path: data.metaData.path });
+      });
+    }
+
+    /**
+     * Otherwise, the layout is left alone and navigation occurs immediately.
+     */
+    return Promise.resolve().then(() => routingStackDelegate.show({ path: data.metaData.path }));
+  }
+
+  render() {
+    const { title, routingStackDelegate, menuItems } = this.props;
+    const { selectedChildKey } = this.state;
+
+    /**
+     * The items using the simplified RoutingMenu menuItem API must be converted into the NavigationSideMenu's API.
+     */
+    const processedMenuItems = RoutingMenu.buildSideMenuItems(menuItems);
+
+    /**
+     * The RoutingMenu then constructs a menuItem that will act as the main page and render the menuItems as child items.
+     * If there is no title and no `showParent` implementation present, the isRootMenu flag is set on the item to hide
+     * the header altogether.
+     */
+    processedMenuItems.push({
+      key: routingMenuRootMenuKey,
+      text: title || '',
+      childKeys: processedMenuItems.map(item => item.key),
+      isRootMenu: !routingStackDelegate.showParent && !title,
```

Comment:
Does this mean submenu's can never have a title?

Reply:
This means that if there is no showParent (i.e. no menu to go back to), and no title is provided, then we hide the header altogether than render a blank one.